lib/matplotlib/_mathtext.py

```diff
@@ -2044,7 +2044,7 @@ def __init__(self):
         p.accentprefixed <<= Suppress(p.bslash) + oneOf(self._accentprefixed)
         p.symbol_name   <<= (
             Combine(p.bslash + oneOf(list(tex2uni)))
```

Comment:
I *think* the problem of asKeyword is that "word boundary" needs a custom definition here (e.g. underscores and digits also separate words).  I this this is configurable on pyparsing's side, but I'd rather just write our own regex (something like `(?![a-zA-Z])`) and be done with it.

Reply:
This sounds like a better approach than trying to do custom keywords.

If you _really_ want to collapse this down to a single `Regex`, then you could use `oneOf `to build a `Regex `for you of just the word choices, but then extract the generated regex pattern into a new `Regex`, something like `Regex(r"\\(" + oneOf(text2uni).pattern + ")(?![a-zA-Z])")`. (untested, ymmv, etc.)

Reply:
tex symbols should not have any metacharacters, so the middle part is probably even just `"|".join(tex2uni)`.

Reply:
Beware of this - `oneOf` also takes care of reordering longer entries before shorter in the event the shorter entry is a leading subset. You _could_  just reverse sort the keys in `tex2uni` by length, but if there are known frequencies to some symbols vs others, and more frequent entries were placed first, `oneOf` would only reorder them to avoid masking entries, whereas sorting by longest-to-shortest would lose some of this priority ordering.

Maybe keep it simple first, just do `"|".join(sorted(tex2uni, key=len, reverse=True))` and then get clever with the frequency-based ordering as a later experiment.

Reply:
Ah, good point, thanks for mentioning that.